internal/cmd/cliopts/flat.go

```diff
@@ -0,0 +1,143 @@
+package cliopts
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mitchellh/reflectwalk"
+	"github.com/spf13/pflag"
+)
+
+func loadFromEnv(target interface{}, opts Options) error {
```

Comment:
This file is probably the most complicated part. `loadFromFile` is a pretty straightforward call to `mapstructure`, but the two in this file (`loadFromEnv`, and `loadFromFlags`) need to do a bit more.

I think it's important to mention that while it is complicated, using `viper` doesn't solve this for us. From my testing `viper` does not work with nested env vars, so this logic would be necessary anyway.

Also the logic in viper to handle flags is pretty rough (ex: https://github.com/spf13/viper/blob/master/viper.go#L1244-L1247). That is some pretty questionable marshaling of values to strings, and then attempting to re-parse them. I think I've found a decent enough way to workaround that problem by using `hookFlagValueSlice` and `HookSetFromString`. An even better option would be to ensure that every `spf13/pflag.Value` implements an `Interface() interface{}` method. That is something we could do by forking `pflag`, and trying to upstream the change.

The approach taken here instead is very similar for both env vars and command line flags.
1. first we build a `map[string]interface{}` from all the values
2. then we iterate over the target struct, descending into any fields that are also structs
3. for each non-struct field , look up a value from the source map we built in 1.